### ISSUE TITLE
Bug: Bluetooth control not working after connection

### ISSUE THREAD

**Issue by mlykotom** [timusus/Shuttle#21]
- **Version**: 1.6.3 (Upgraded)
- **Device**: OnePlus 3T (was happening on OnePlus One as well)
- **OS version**: Android 7.0 (was happening on previous versions as well)
- **Description**: After connecting phone to bluetooth device (tried with headphones and bluetooth in a car), it's not possible to control the player. Simply any playback controls (play/pause, next/prev, is not responding). But after you play a song from widget/app, then it's possible to control it via bluetooth. If you first play a song and later connect to bluetooth, it's possible to control it.

- **Outcome**: I assume that the reason it's happening is, that when you connect bluetooth device and app was not running, the broadcast receiver for bluetooth controls is not registered, therefore it can't control the app.

**Comment by Techman-** [comment]
Same thing happens to me. If the app is already started and running, my bluetooth stereo (car) seems to connect just fine and I can start playing music. If the app isn't started then the controls don't work, and worse, some other app like VLC will start and play whatever music it chooses from (or the last playlist if available).

I hope this gets sorted soon :)

**Comment by timusus** [comment]
I think the solution to this problem is to host the MediaButtonReceiver in a service, which runs at boot.

The downside is it requires an additional permission `<uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED" />` which I've been avoiding in case it scares people.

**Comment by Techman-** [comment]
As long as permissions are given an adequate explanation, I see no problem with it. It's actually better this way because sometimes apps have to request permissions that themselves sound scary but actually serve a very specific purpose.

**Comment by timusus** [comment]
More specifically talking about the average Play Store user, who is wondering why the hell their music player needs to know when the device is booted.

Anyway, that's just the reason I've been avodiing this - and it's not a great one. So I'm happy to put this back on the cards.

**Comment by Techman-** [comment]
Looks like [VLC](https://code.videolan.org/videolan/vlc-android/blob/master/vlc-android/AndroidManifest.xml#L48) already uses this permission so I don't think it's out of bounds to use it here.

**Comment by timusus** [comment]
Yeah, it's definitely not out of bounds, it's just a consideration.

**Comment by vedard** [comment]
Version: 1.6.4
Device: Samsung edge S7 and HTC m8
Android: 6

Resume playback when Bluetooth is connected is not working when the app is completely closed. Sometimes other app like Spotify or Play music start.

It's probably the same problem.
